include/cuco/utility/fast_int.cuh

```diff
@@ -150,10 +150,16 @@ struct fast_int {
     return rhs.mulhi(rhs.magic_, mul) >> rhs.shift_;
   }
 
+  template <typename Rhs>
+  friend __host__ __device__ constexpr value_type operator/(fast_int const& lhs, Rhs rhs) noexcept
+  {
+    return lhs.value() / static_cast<value_type>(rhs);
```

Comment:
Is the static cast required? I would instead let the compiler figure out the result type depending on `Rhs` rather than enforcing the `value_type`.